openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -284,7 +284,18 @@ def verify(self, user_id):
             ).unsign_object(self.signature, serializer=signing.JSONSerializer, max_age=settings.SESSION_COOKIE_AGE)
 
             if unsigned_data == self._compute_digest(user_id):
+                set_custom_attribute('SafeCookieData.verified.alg', "default")
                 return True
+            # Fallback when DEFAULT_HASHING_ALGORITHM is upgraded to 256
+            if settings.DEFAULT_HASHING_ALGORITHM != 'sha1':
+                # if not verified from DEFAULT_HASHING_ALGORITHM, try to verify from sha1
+                unsigned_data = signing.TimestampSigner(
+                    salt=self.key_salt, algorithm="sha1"
+                ).unsign_object(self.signature, serializer=signing.JSONSerializer, max_age=settings.SESSION_COOKIE_AGE)
+
+                if unsigned_data == self._compute_digest(user_id):
+                    set_custom_attribute('SafeCookieData.verified.alg', "fallback")
```

Comment:
Consider adding a "Temporary Hash Change Rollout" dashboard in New Relic with charts like:
```
SELECT count(*) FROM Transaction WHERE SafeCookieData.verified.alg IS NOT NULL FACET SafeCookieData.verified.alg TIMESERIES MAX
```
Note: I'm not sure if New Relic will require quotes with the attribute name with `.`, so you could also rename it to `SafeCookieData_verified_alg`, or you may need to use ``` in the NRQL.